### DISCUSSION TITLE
How to represent absolute paths to binaries?

### DISCUSSION THREAD

**Discussion by m3thm4th** [gmodena/nix-flatpak#99] [Q&A]
For example the absolute path to Thunderbird from the Nix repo would be `${pkgs.thunderbird}/bin/thunderbird`
What would be the equivalent for the flatpak `org.mozilla.Thunderbird`?

**Comment by gmodena** [comment]
What are you trying to achieve?

The typical path for a Flatpak application will be within the `~/.local/share/flatpak/app` directory for `user` installations,  or  `/var/lib/flatpak/app` for `system` wide ones.

Each application is built and run in an isolated environment (the [sanbox](https://docs.flatpak.org/en/latest/basic-concepts.html)), installed side-by-side  nix store. 

`nix` will not be aware of those environments, but if you want to find the absolute path for an executable you can query `flatpak` directly with:

 `flatpak info --show-location org.mozilla.Thunderbird`.

**Comment by m3thm4th** [comment]
So basically it will be `/var/lib/flatpak/app` because they are non-generational, the path variable would be needed with other projects like [this one](https://github.com/GermanBread/declarative-flatpak) where Flatpaks are generational, correct?

**Comment by gmodena** [comment]
I'm not sure how the implementation you linked handles things these days, but the path isn't related to the fact that nix-flatpak is not generational.

Flatpak (upstream) uses its own directory layout to store its application sandboxes outside of the Nix store. Nix, and its tooling, won't be aware of those binaries.
